components/prism-sas.js

```diff
@@ -10,25 +36,137 @@ Prism.languages.sas = {
-			lookbehind: true
+	'procSQL': {
+		pattern: /(^|\n)(?:proc\s+sql(?:\s+[\w|=]+)?);([\s\S]+(?:\r?\n|\r)(?=((?:proc\s+\w+|quit|run|data(?!\=));))((?:proc\s+\w+|quit|run|data(?!\=));)|(?:\r?\n|\r)[\s\S]+;)/im,
+		inside: {
+			'fullStep': {
+				pattern: /(^|\n)(?:proc\s+sql(?:\s+[\w|=]+)?);/i,
+				inside: {
+					'step': step,
+					'argument': {
+						pattern: /\w+(?==)/,
+						alias: 'keyword'
+					},
+					'operator': /=/,
+					'punctuation': /;/,
+					'number': numberPattern,
+					'numeric-constant': numericConstant,
+					'string': string
+				}
+			},
+			'step': step,
+			rest: Prism.languages.sql
+		}
+	},
+	/*Special keywords within macros*/
+	'macro-keyword': {
+		pattern: /((^|[\s])=?)%(?:ABORT|BQUOTE|BY|CMS|COPY|DISPLAY|DO|ELSE|END|EVAL|GLOBAL|GO|GOTO|IF|INC|INCLUDE|INDEX|INPUT|KTRIM|LENGTH|LET|LIST|LOCAL|NRBQUOTE|NRQUOTE|NRSTR|PUT|QKTRIM|QSCAN|QSUBSTR|QSYSFUNC|QUOTE|QUPCASE|RETURN|RUN|SCAN|STR|SUBSTR|SUPERQ|SYMDEL|SYMGLOBL|SYMLOCAL|SYMEXIST|SYSCALL|SYSEVALF|SYSEXEC|SYSFUNC|SYSGET|SYSRPUT|THEN|TO|TSO|UNQUOTE|UNTIL|UPCASE|WHILE|WINDOW)\b/i,
+		lookbehind: true,
+		alias: 'keyword'
+	},
+	'macro-declaration': {
+		pattern: /^%macro[\s\S]+?;/im,
+		inside: {
+			keyword: /%macro/i,
+		}
+	},
+	'macro-end': {
+		pattern: /^%mend[\s\S]+?;/im,
+		inside: {
+			keyword: /%mend/i,
+		}
+	},
+	/*%_zscore(headcir, _lhc, _mhc, _shc, headcz, headcpct, _Fheadcz); */
+	'macro': {
+		pattern: /%_\w+(?=\()/,
+		alias: 'keyword'
+	},
+	'input': {
+		pattern: /\binput\s+[-\w\s/*.$&]+;/i,
+		inside:{
+			'input':{
+				alias: 'keyword',
+				pattern: /^input/i,
+			},
+			'comment': comment,
+			'number': numberPattern,
+			'numeric-constant': numericConstant
+		}
+	},
+	'comment': comment,
+	'options': {
+		pattern: /^options[-'"|/\\<>*+=()\w\s]*;/im,
+		inside: {
+			'options': {
+				alias: 'keyword',
+				pattern: /(^options)/i,
+			},
+			'equals': {
+				pattern: /=/,
+				alias: 'operator'
+			},
+			'parentheses': {
+				pattern: /(\()([A-Z]+)(\))/i,
+			},
+			'arg': {
+				pattern: /([A-Z]+)/i,
+				alias: 'keyword'
+			},
+			'number': numberPattern,
+			'numeric-constant': numericConstant,
+			'string': string
 		},
-		/\/\*[\s\S]+?\*\//
-	],
+	},
+	'function': {
+		pattern: /%?\w+(?=\()/,
+		alias: 'keyword'
+	},
+	'format': {
+		pattern: /(\b)(?:((format)|(put))(?:=?))([\w\'\$\.]+)/im,
+		inside: {
+			'keyword': /(format|put)(?=\=)/i,
+
+			'equals': {
+						pattern: /=/,
+						alias: 'operator'
+					},
+			'format': {
+					pattern: /(\w|\$\d)+\.\d?/i,
+					alias: 'number',
+				}
+		}
+	},
+	'altformat': {
+		pattern: /\b(?:(?:format|put)\s+)[\w']+(\s+[$.\w]+)+;/i,
+		inside: {
+			'keyword': /(format|put)/i,
+			'format': {
+					pattern: /(\w|\$)+\.\d?/,
+					alias: 'number',
+				},
+				'punctuation': /;/
+		}
+	},
+	'numeric-constant': numericConstant,
 	'datetime': {
 		// '1jan2013'd, '9:25:19pm't, '18jan2003:9:27:05am'dt
-		pattern: /'[^']+'(?:dt?|t)\b/i,
+		pattern: /(["'])(?:.+)\1(?:dt?|t)/,
```

Comment:
`["']` ⊂ `.`. As a consequense, your pattern will match two dates after another as one date.
```suggestion
		pattern: /(["'])(?:(?!\1).)+\1(?:dt?|t)/,
```

Reply:
Also, why did you remove the `i` flag?

Reply:
"dt" and "t" are case sensitive in SAS. I tested in SAS Studio.

Reply:
Ok.